File: library/src/main/java/com/dazhi/libroot/root/RootVmFragment.java
package com.dazhi.libroot.root;

/**
 * 功能：
 * 描述：
 * 作者：WangZezhi
 * 邮箱：wangzezhi528@163.com
 * 创建日期：2019/2/27 14:27
 * 修改日期：2019/2/27 14:27
 */
public abstract class RootVmFragment<T extends RootViewModel> extends RootFragment {
    protected T vm; // vm=ViewModelProviders.of(activity).get(VmDebug.class);

    @Override
    protected void initViewAndDataAndEvent() {
        vm.attachUiView(this);
    }

}

File: library/src/main/java/com/dazhi/libroot/root/RootVmFragment.kt
package com.dazhi.libroot.root

import androidx.viewbinding.ViewBinding
import com.dazhi.libroot.inte.IRootView

/**
 * 功能：
 * 描述：
 * 作者：WangZezhi
 * 邮箱：wangzezhi528@163.com
 * 创建日期：2019/2/27 14:27
 * 修改日期：2019/2/27 14:27
 */
abstract class RootVmFragment<T: RootViewModel<IRootView>, E: ViewBinding> : RootFragment<E>() {
    // vm=ViewModelProviders.of(activity).get(VmXXX.class);
    @Suppress("MemberVisibilityCanBePrivate")
    protected lateinit var vm: T
    protected abstract fun initVm(): T

    override fun initConfig() {
        vm = initVm()
    }
    override fun initViewAndDataAndEvent() {
        vm.attachUiView(this)
    }
}